auth/api/iam/openid4vp.go

```diff
@@ -193,44 +217,53 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 // there are way more error conditions that listed at: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#name-error-response
 // missing or invalid parameters are all mapped to invalid_request
 // any operation that fails is mapped to server_error, this includes unreachable or broken backends.
-func (r Wrapper) handleAuthorizeRequestFromVerifier(ctx context.Context, walletDID did.DID, params map[string]string) (HandleAuthorizeRequestResponseObject, error) {
+func (r Wrapper) handleAuthorizeRequestFromVerifier(ctx context.Context, walletDID did.DID, params map[string]interface{}) (HandleAuthorizeRequestResponseObject, error) {
 	responseMode := params[responseModeParam]
 	if responseMode != responseModeDirectPost {
 		return nil, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "invalid response_mode parameter"}
 	}
 
 	// check the response URL because later errors will redirect to this URL
-	responseURI, responseOK := params[responseURIParam]
+	responseURI, responseOK := params[responseURIParam].(string)
 	if !responseOK {
 		return nil, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "missing response_uri parameter"}
 	}
 	// we now have a valid responseURI, if we also have a clientState then the verifier can also redirect back to the original caller using its client state
-	state := params[oauth.StateParam]
+	state, ok := params[oauth.StateParam].(string)
+	if !ok {
+		return nil, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "missing state parameter"}
+	}
+	// do a quick check for any missing string values where a conversion would blow up
+	for _, param := range []string{clientMetadataURIParam, presentationDefUriParam, oauth.ClientIDParam} {
+		if _, ok := params[param].(string); !ok {
+			return r.sendAndHandleDirectPostError(ctx, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: fmt.Sprintf("missing %s parameter", param)}, responseURI, state)
+		}
+	}
 
 	clientIDScheme := params[clientIDSchemeParam]
 	if clientIDScheme != didScheme {
 		return r.sendAndHandleDirectPostError(ctx, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "invalid client_id_scheme parameter"}, responseURI, state)
 	}
 
-	verifierID := params[oauth.ClientIDParam]
+	verifierID := params[oauth.ClientIDParam].(string)
 	// the verifier must be a did:web
 	verifierDID, err := did.ParseDID(verifierID)
 	if err != nil || verifierDID.Method != "web" {
 		return r.sendAndHandleDirectPostError(ctx, oauth.OAuth2Error{Code: oauth.InvalidRequest, Description: "invalid client_id parameter (only did:web is supported)"}, responseURI, state)
 	}
 
-	nonce, ok := params[nonceParam]
+	nonce, ok := params[nonceParam].(string)
```

Comment:
with so many string casts, better take another approach. Maybe make params a type with a `String(key string) string` function that returns the param as string, if we really need it to be `map[string]interface{}`

Reply:
fixed with  `singleStringParam`